### PR TITLE
ref(ts): Convert InternalStatChart

### PR DESCRIPTION
**Pull request by davidenwang**
refactor to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/internalStatChart.tsx [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
-  static defaultProps = {
-    height: 150,
-  };
+type Props = {
+  api: Client;
+  since: number;
+  resolution: string;
+  stat: string;
+  label: string;
+} & Partial<typeof defaultProps>;
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/components/internalStatChart.tsx:19]: I know that there is some debate over how to properly and @EvanPurkhiser is in favor of limiting use of default props. In that case would it be best to explicitly pass in `height = {150}` for all parent components that render this `InternalStatChart` component?

Comment (taylangocmen) [reply src/sentry/static/sentry/app/components/internalStatChart.tsx:19]: You can pass `height={height || 150}` to `MiniBarChart` as well imo

Comment (taylangocmen) [reply src/sentry/static/sentry/app/components/internalStatChart.tsx:19]: And have `height?: number` in `type Props`

**Review by taylangocmen — APPROVED** [review]
Other than the `defaultProps` lgtm

File: src/sentry/static/sentry/app/components/internalStatChart.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/internalStatChart.tsx:19]: I still prefer the defaultProps 😐 let's discuss it in the next tsc 😉

**Review by priscilawebdev — APPROVED** [review]
Good Job!!

File: src/sentry/static/sentry/app/components/internalStatChart.tsx (hunk 1 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/components/internalStatChart.tsx:19]: following discussions from the tsc i'm gonna choose to make height an optional prop and then default it to 150 if it's null.

**Pull request merged**